app/grandchallenge/components/form_fields.py

```diff
@@ -144,42 +148,39 @@ def get_json_field(self):
         )
 
     def get_file_field(self):
-        key = f"value_type_{INTERFACE_FORM_FIELD_PREFIX}{self.instance.slug}"
-        if key in self.form_data.keys():
-            value_type = self.form_data[key]
-        elif self.civs_for_user_for_interface.exists():
-            value_type = "civ"
-        else:
-            value_type = "uuid"
-
-        if value_type == "uuid":
-            extra_help = f"{file_upload_text} {self.instance.file_extension}"
-            return ModelChoiceField(
-                queryset=get_objects_for_user(
-                    self.user,
-                    "uploads.change_userupload",
-                ).filter(status=UserUpload.StatusChoices.COMPLETED),
-                widget=UserUploadSingleWidget(
-                    allowed_file_types=self.instance.allowed_file_types,
-                ),
-                help_text=_join_with_br(self.help_text, extra_help),
-                **self.kwargs,
-            )
-        elif value_type == "civ":
-            file_upload_link = reverse(
-                "components:file-upload",
-                kwargs={
-                    "interface_slug": self.instance.slug,
-                },
-            )
+        current_value = None
 
-            return ModelChoiceField(
-                queryset=self.civs_for_user_for_interface,
-                widget=SelectUploadWidget(
-                    attrs={"upload_link": file_upload_link}
-                ),
-                **self.kwargs,
-            )
+        if self.initial:
+            if isinstance(self.initial, ComponentInterfaceValue):
+                current_value = self.initial.file
+            elif (
+                isinstance(self.initial, int) or self.initial.isdigit()
+            ) and ComponentInterfaceValue.objects.filter(
+                pk=self.initial
+            ).exists():
+                current_value = ComponentInterfaceValue.objects.get(
+                    pk=self.initial
+                )
+            elif UserUpload.objects.filter(pk=self.initial).exists():
+                current_value = UserUpload.objects.get(pk=self.initial)
+            else:
+                raise RuntimeError(
+                    f"Unknown type for initial value: {self.initial}"
+                )
+
+        self.kwargs["widget"] = FlexibleFileWidget(
+            help_text=self.help_text,
+            user=self.user,
+            current_value=current_value,
+            # also passing the CIV as current value here so that we can
+            # show the image name to the user rather than its pk
```

Comment:
Maybe this comment can go if you add comments above. 

```suggestion
            # also passing the CIV as current value here so that we can
            # show the filename to the user rather than its pk
```